File: socialview/src/main/java/io/github/hendraanggrian/socialview/SocialEditText.java
package io.github.hendraanggrian.socialview;

import android.content.Context;
import android.os.Build;
import android.support.annotation.RequiresApi;
import android.util.AttributeSet;
import android.widget.EditText;

/**
 * @author Hendra Anggrian (hendraanggrian@gmail.com)
 */
public class SocialEditText extends EditText {

    public SocialEditText(Context context) {
        super(context);
    }

    public SocialEditText(Context context, AttributeSet attrs) {
        super(context, attrs);
    }

    public SocialEditText(Context context, AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
    }

    @RequiresApi(api = Build.VERSION_CODES.LOLLIPOP)
    public SocialEditText(Context context, AttributeSet attrs, int defStyleAttr, int defStyleRes) {
        super(context, attrs, defStyleAttr, defStyleRes);
    }
}

File: socialview/src/com/hendraanggrian/widget/SocialEditText.kt
package com.hendraanggrian.widget

import android.content.Context
import android.support.v7.widget.AppCompatEditText
import android.util.AttributeSet
import com.hendraanggrian.socialview.SocialView
import com.hendraanggrian.socialview.SocialViewImpl

/**
 * [android.widget.EditText] with hashtag, mention, and hyperlink support.
 *
 * @see SocialView
 * @see SocialViewImpl
 */
class SocialEditText @JvmOverloads constructor(
        context: Context,
        attrs: AttributeSet? = null,
        defStyleAttr: Int = android.support.v7.appcompat.R.attr.editTextStyle,
        impl: SocialViewImpl = SocialViewImpl()
) : AppCompatEditText(context, attrs, defStyleAttr), SocialView by impl {

    init {
        impl.init(this, attrs)
    }
}